app/src/functions/notify-endpoint-usage/app.ts

```diff
@@ -55,17 +57,42 @@ function getRelaysUsed(networkData: QueryAppResponse[], influxData: GetUsageData
   return applicationsData
 }
 
-exports.handler = async () => {
-  const usage = await getUsageData();
+async function getNetworkData(influxData: GetUsageDataQuery[]): Promise<QueryAppResponse[]> {
+  const sleep = (seconds: number, factor: number) => new Promise(resolve => setTimeout(resolve, (seconds ** factor) * 1000))
+
+  const networkApps: QueryAppResponse[] = []
+  const failedApps: GetUsageDataQuery[] = []
 
-  const networkApps = Promise.allSettled(
-    usage.map((app) => getApplicationNetworkData(app.applicationPublicKey))
-  );
+  for (let i = 0; i < maxRetries; i++) {
+    if (i > 0) {
+      await sleep(2, i)
```

Comment:
I don't get this. Is this just a way to stop before the next try? I really think a `setTimeout` wrapping the whole retry function would work better, however this is probably good enough too.